velox/exec/HashBuild.cpp

```diff
@@ -1203,4 +1235,10 @@ void HashBuildSpiller::extractSpill(
         rows.data(), rows.size(), false, false, result->childAt(types.size()));
   }
 }
+
+bool HashBuild::abandonBuildNoDupHashEarly(int64_t numDistinct) const {
+  VELOX_CHECK(dropDuplicates_);
+  return numHashInputRows_ > abandonBuildNoDupHashMinRows_ &&
+      100 * numDistinct / numHashInputRows_ >= abandonBuildNoDupHashMinPct_;
```

Comment:
numDistinct / numHashInputRows_ >=  abandonBuildNoDupHashMinPct_ / 100 for overflow case